test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,233 @@
+*** Settings ***
+Documentation       Host IP and Certificate change test Suite
+
+Resource            ../resources/common.resource
+Resource            ../resources/systemd.resource
+Resource            ../resources/microshift-config.resource
+Resource            ../resources/microshift-process.resource
+
+Library             OperatingSystem
+Library             SSHLibrary
+Library             BuiltIn
+Library             String
+Library             random
+
+Suite Setup         Setup Suite
+Suite Teardown      Teardown Suite
+
+Test Tags           restart    slow
+
+*** Variables ***
+${HOSTS_FILE_PATH}  /etc/hosts
+${EXTERNAL_SERVER_CERT}    /var/lib/microshift/certs/kube-apiserver-external-signer/kube-external-serving/server.crt
+
+
+*** Keywords ***
+Generate New IP
+    ${hostip}     ${rc1}=    Execute Command    microshift show-config| grep -i "nodeIP:" | awk '{print $2}'    sudo=True   return_stdout=True    return_rc=True
+    Should Be Equal As Integers    ${rc1}    0
+    ${gateway}    ${rc2}=    Execute Command    ip route show | grep -i ${hostip} | grep -i default | awk '{print $3}'    return_stdout=True    return_rc=True
+    Should Be Equal As Integers    ${rc2}    0
+    ${device}     ${rc3}=    Execute Command    ip route show | grep -i ${hostip} | grep -i default | awk '{print $5}'    return_stdout=True    return_rc=True
+    Should Be Equal As Integers    ${rc3}    0
+
+    FOR    ${last_digit}    IN RANGE    2    255
+        ${digit}    Evaluate    int($hostip.split('.')[-1])
+        ${new_host_ip}    Evaluate    '${hostip}'.rsplit('.', 1)[0] + '.' + str(${last_digit})
+        Log    Attempt ${last_digit}: Generated IP: ${new_host_ip}
+        ${ping_output}    Execute Command    ping -c 1 ${new_host_ip} | grep -i "100% packet loss"| wc -l
+        Run Keyword If    ${ping_output} > 0    Exit For Loop
+    END
```

Comment:
This code still does not solve a problem of the potential future conflict of the IPs. Since the IP is changed inside the VM, the hypervisor does not know about it and it can reuse this IP creating a hard-to-debug problem in the local network.

P.S. To check `ping` results, it's sufficient to check the return code without parsing its output.

Reply:
Updated